### DISCUSSION TITLE
Put, but for putting things into the command line after selection

### DISCUSSION THREAD

**Discussion by brumm** [junegunn/fzf#4101] [Q&A]
I'm using fzf to select from a list of commands and run them with `become`. Some commands can't be run on their own and need an argument. I'd like to add a new binding which would end fzf and print the selection to my console input prompt, so I can add the required parameters.

Is that something I should do with some existing tooling or could this become part of fzf? I've tried out fish's `commandline` command but it can't run in non-interactive mode and is not portable.

**Comment by LangLangBart** [comment]
The easiest way would be to place your command into a command substitution and then pass it to `commandline`.

```sh
commandline $(fzf --with-shell 'fish -c' --bind 'start:become:echo $FISH_VERSION')
```



Related Issue:
- [Result of 'fzf' pasted to prompt/command line instead of stdout · Issue #1849 · junegunn/fzf · GitHub](https://github.com/junegunn/fzf/issues/1849#issuecomment-765651686)
